src/apps/watchfaces/OswAppWatchfaceMonotimer.cpp

```diff
@@ -0,0 +1,146 @@
+
+#include "./apps/watchfaces/OswAppWatchfaceMonotimer.h"
+// #define GIF_BG
+
+#ifdef ANIMATION
+#include <anim_matrix.h>
+#endif
+
+#include <gfx_util.h>
+#include <osw_app.h>
+#include <osw_config.h>
+#include <osw_config_keys.h>
+#include <osw_hal.h>
+
+#ifdef GIF_BG
+#include "./apps/_experiments/gif_player.h"
+#endif
+
+void OswAppWatchfaceMonotimer::drawNShiftedTicks(Graphics2D *gfx, uint8_t cx, uint8_t cy, uint8_t r1, uint8_t r2, uint8_t nTicks, float shift, uint16_t color) {
+    const float deltaAngle = 360.0 / nTicks;
+    for(uint16_t i=0; i<nTicks; ++i) {
+        gfx->drawTick(cx, cy, r1, r2, (i * deltaAngle) + shift, color);
+    }
+}
+
+void OswAppWatchfaceMonotimer::drawNShiftedMaskedTicks(Graphics2D *gfx, uint8_t cx, uint8_t cy, uint8_t r1, uint8_t r2, uint8_t nTicks, float shift, uint16_t m, uint16_t color) {
+    const float deltaAngle = 360.0 / nTicks;
+    for(uint16_t i=0; i<nTicks; ++i) {
+        if(i%m) {
+            gfx->drawTick(cx, cy, r1, r2, (i * deltaAngle) + shift, color);
+        }
+    }
+}
+
+void OswAppWatchfaceMonotimer::drawWatch() {
+    OswHal* hal = OswHal::getInstance();
+
+    // hours
+    hal->gfx()->drawNTicks(120, 120, 117, 100, 12, ui->getForegroundColor());
+    // 30 minutes
+    drawNShiftedTicks(hal->gfx(), 120, 120, 117, 105, 12, 360.0/24.0, ui->getForegroundColor());
+    // 15 minutes
+    drawNShiftedTicks(hal->gfx(), 120, 120, 110, 105, 24, 360.0/48.0, ui->getForegroundColor());
+    // 5 minutes
+    drawNShiftedMaskedTicks(hal->gfx(), 120, 120, 109, 108, 144, 0.0, 3, ui->getForegroundColor());
+
+    // hour labels
+    hal->gfx()->setTextSize(2);
+    hal->gfx()->setTextMiddleAligned();
+
+    static const uint8_t positions[]= {
+        155, 45, 	// 01
+        180, 75, 	// 02
+        200, 120,	// 03
+        183, 165, 	// 04
+        155, 197, 	// 05
+        110, 210, 	// 06
+        65, 197, 	// 07
+        37, 165, 	// 08
+        23, 120, 	// 09
+        37, 75, 	// 10
+        65, 45, 	// 11
+        110, 30 	// 12
+    };
+
+    for(uint8_t i=0; i<12; ++i) {
+        hal->gfx()->setTextCursor(positions[i<<1], positions[(i<<1)|1]);
+        hal->gfx()->printDecimal(i+1, 2);
+    }
+
+#if OSW_PLATFORM_ENVIRONMENT_ACCELEROMETER == 1
+    const uint32_t steps = hal->environment->getStepsToday();
```

Comment:
Why did you use `const`?

Reply:
Clarification that this value will not be modified. Shall I remove it?

Reply:
Yes, I don't think we need this part.